### PR TITLE
Improve snowfall

### PR DESCRIPTION
**Pull request by GreenXenith**
* More snowflake styles
* Overcast sky
* Better fall behavior
* No packet increase*

Also rewrites credits section.

*There are 14 extra textures to send on join, this could be changed though with a tile sheet.

### CODE REVIEW

**Comment by Thomas--S** [comment]
I think this is definitely an improvement. There are two issues I found during testing this:
1) Because of the reduced velocity players can't see any snow while sprinting. I'd suggest to keep the old y-axis velocity values.
2) Some of the snowflake textures are very similiar, I don't think that most players will even notice a difference for some textures. I think the number of textures could be reduced to 7 or 8.

**Comment by GreenXenith** [comment]
1. I sped up the velocity by 2 and sped up the globalstep by 0.2 -- while sprinting at full (default) speed I was able to see a few snowflakes, so I assume it is fast enough for CTF sprint to see (I ~~cba to~~ can't test at the moment).
2. I just copied the snowflakes from snowdrift (which originally had 10, but I think I added 5) and left them as-is, so yeah, there were really more than needed. I have now packed the snowflakes into a single texture of 8 snowflakes, which is much cleaner and sane.

File: snow/init.lua [hunk 1]
```diff
@@ -2,47 +2,73 @@
...
 		spawner_step = 0
 
 		for _, player in pairs(minetest.get_connected_players()) do
+			-- Spawn snow
 			spawn_particles(player)
+
+			-- Update sky (stolen from snowdrift https://github.com/paramat/snowdrift/blob/master/init.lua#L132-L151)
+			local NISVAL = 10 -- Overcast sky RGB value at night (brightness)
+			local DASVAL = 90 -- Overcast sky RGB value in daytime (brightness)
+			local difsval = DASVAL - NISVAL
```
Comment (ClobberXD) [inline snow/init.lua]: I think you've placed all the code from snowdrift in one place (for ease of attribution?). But these three variables seem to be constants - I suggest placing these at the top of the file (defined outside all scopes), for better performance.

Comment (GreenXenith) [reply snow/init.lua]: They were originally at the top, but yes, I moved them to stay grouped with the rest of the snowdrift code. I suppose it's minor, so they have been moved accordingly.

**Comment by ClobberXD** [comment]
This looks really good!

screenshot_20191202_112936
screenshot_20191202_112929

**Review by ClobberXD — APPROVED** [review]
Tested. Everything works as expected and code looks good as well.

**Pull request merged**